features/256-auth-gear-interaction-overview-in-next/api.md

```diff
@@ -172,25 +176,38 @@ Followings are hooks of auth actions:
 | -------- | -------- | 
 | `signup` | `before_signup_sync(user, context)`<br/>`before_signup(user, context)`<br/>`after_signup_sync(user, context)`<br/>`after_signup(user, context)`<br/> |
 | `login` | `before_login_sync(user, context)`<br/>`before_login(user, context)`<br/>`after_login_sync(user, context)`<br/>`after_login(user, context)` |
-| `disable` | `before_disable_sync(user, context)`<br/>`before_disable(user, context)`<br/>`after_disable_sync(user, context)`<br/>`after_disable(user, context)` |
-| `enable` | `before_enable_sync(user, context)`<br/>`before_enable(user, context)`<br/>`after_enable_sync(user, context)`<br/>`after_enable(user, context)` |
-| `roles` | `before_change_roles_sync(user, context)`<br/>`before_change_roles(user, context)`<br/>`after_change_roles_sync(user, context)`<br/>`after_change_roles(user, context)` |
 | `logout` | `before_logout_sync(user, context)`<br/>`before_logout(user, context)`<br/>`after_logout_sync(user, context)`<br/>`after_logout(user, context)` |
-| `change_password` | `before_change_password_sync(user, context)`<br/>`before_change_password(user, context)`<br/>`after_change_password_sync(user, context)`<br/>`after_change_password(user, context)` |
-| `reset_password` | `before_reset_password_sync(user, context)`<br/>`before_reset_password(user, context)`<br/>`after_reset_password_sync(user, context)`<br/>`after_reset_password(user, context)` |
-| `verify` | `before_verify_sync(user, context)`<br/>`before_verify(user, context)`<br/>`after_verify_sync(user, context)`<br/>`after_verify(user, context)` |
-| `unverify` | `before_unverify_sync(user, context)`<br/>`before_unverify(user, context)`<br/>`after_unverify_sync(user, context)`<br/>`after_unverify(user, context)` |
-| `update_metadata` | `before_update_metadata_sync(user, original_user, context)`<br/>`before_update_metadata(user, original_user, context)`<br/>`after_update_metadata_sync(user, original_user, context)`<br/>`after_update_metadata(user, original_user, context)` | 
+| `roles` | `before_roles_changed_sync(user, original_user, context)`<br/>`before_roles_changed(user, original_user, context)`<br/>`after_roles_changed_sync(user, original_user, context)`<br/>`after_roles_changed(user, original_user, context)` |
+| `enable` | `before_enable_changed_sync(user, original_user, context)`<br/>`before_enable_changed(user, original_user, context)`<br/>`after_enable_changed_sync(user, original_user, context)`<br/>`after_enable_changed(user, original_user, context)` |
```

Comment:
naming is confusion. Either:

1. Keep `[before|after]_enabled` / `[before|after]_disabled`, or
1. `[before|after]_enable_status_changed`.

I prefer the former.